validator/main.py

```diff
@@ -231,27 +239,30 @@ def validate(
         if bool(valid_topics.intersection(invalid_topics)):
             raise ValueError("A topic cannot be valid and invalid at the same time.")
 
-        # Add 'other' to the invalid topics list
-        if "other" not in invalid_topics:
-            invalid_topics.add("other")
-
-        # Combine valid and invalid topics
-        candidate_topics = valid_topics.union(invalid_topics)
-
         # Check which model(s) to use
         if self._disable_classifier and self._disable_llm:  # Error, no model set
             raise ValueError("Either classifier or llm must be enabled.")
         elif (
             not self._disable_classifier and not self._disable_llm
-        ):  # Use ensemble (Zero-Shot + Ensemble)
-            return self.get_topic_ensemble(value, list(candidate_topics))
+        ):  
+            if invalid_topics:# Use ensemble (Zero-Shot + Ensemble)
+                return self.get_topic_ensemble(value, list(invalid_topics))
```

Comment:
Instead of returning Pass/FailResults within the `get_topic_*` functions, can you change the scope of those functions so they only return the topics present in the text, and the final processing of whether the topics present in the text should be allowed/disallowed and the Pass/FailResult generation should happen in one place within the `validate` fn.

Right now, the Pass/FailResult generation being scattered across various methods impacts readability and makes it hard to understand the code.